app/global/reducers/questionGroups/reducer.js

```diff
@@ -77,8 +77,14 @@ const questionGroupsReducer = (state = initialState, { type, payload }) =>
         break;
       }
       case UPDATE_QUESTION_GROUP_REQUEST: {
-        const { groupId, data } = payload;
-        updateItemById(draft.groups, groupId, data);
+        const index = state.groups.findIndex(
+          ({ id }) => id === payload.groupId,
+        );
+
+        draft.groups[index] = merge(
+          cloneDeep(state.groups[index]),
+          payload.data,
+        );
```

Comment:
I see that you came back with to the old solution. Was `updateItemById ` causing bugs? I see that success and error actions still use it